File: AspNetAngular1/Data/Entity/Product.cs
namespace AspNetAngular1.Data.Entity
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;
    using System.Data.Entity.Spatial;

    public partial class Product
    {
        public int ProductID { get; set; }

        public Guid ProductGUID { get; set; }

        [Required]
        [StringLength(50)]
        public string ProductName { get; set; }

        public int Quantity { get; set; }
    }
}

File: AspNetAngular1/App/Product/Product.ts
namespace MyApp {
    export namespace Domain {

        export interface IProduct {
            ProductGUID: string;
            ProductName: string;
            Quantity: number;
        }

        export class Product implements IProduct {
            constructor() { }

            public ProductGUID: string;
            public ProductName: string;
            public Quantity: number;
        }

    }
}